ext/menoh_native/menoh_ruby.c

```diff
@@ -141,29 +144,17 @@ static VALUE wrap_model_init(VALUE self, VALUE vonnx, VALUE option) {
     int32_t dims_length =
         NUM2INT(rb_funcall(vdims, rb_intern("length"), 0, NULL));
 
-    switch (dims_length) {
-      case 2:
-        ERROR_CHECK(
-            menoh_variable_profile_table_builder_add_input_profile_dims_2(
-                getModel(self)->vpt_builder, StringValueCStr(vname),
-                menoh_dtype_float, NUM2INT(rb_ary_entry(vdims, 0)),
-                NUM2INT(rb_ary_entry(vdims, 1))),
-            rb_eStandardError);
-        break;
-      case 4:
-        ERROR_CHECK(
-            menoh_variable_profile_table_builder_add_input_profile_dims_4(
-                getModel(self)->vpt_builder, StringValueCStr(vname),
-                menoh_dtype_float, NUM2INT(rb_ary_entry(vdims, 0)),
-                NUM2INT(rb_ary_entry(vdims, 1)),
-                NUM2INT(rb_ary_entry(vdims, 2)),
-                NUM2INT(rb_ary_entry(vdims, 3))),
-            rb_eStandardError);
-        break;
-      default:
-        rb_raise(rb_eStandardError, "invalid dimension length");
-        return Qnil;
+    getModel(self)->dims = (int32_t *)ruby_xmalloc(sizeof(int32_t) * dims_length);
```

Comment:
Previously allocated `getModel(self)->dims` is not free'ed, so that it causes memory leak.

I recommend simply using `alloca` to allocate temporal buffer instead of `ruby_xmalloc` and using local variable instead of adding `dims` field to `menohModel`.

Reply:
I agree with your recommendation. fixed.